drv/i2c-devices/src/mwocp68.rs

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! MWOCP68-3600 Murata power shelf
+
+use crate::{CurrentSensor, Validate, VoltageSensor};
+use core::cell::Cell;
+use drv_i2c_api::*;
+use pmbus::commands::mwocp68::*;
+use pmbus::commands::CommandCode;
+use pmbus::units::{Celsius, Rpm};
+use pmbus::*;
+use userlib::units::{Amperes, Volts};
+
+pub struct Mwocp68 {
+    device: I2cDevice,
+
+    /// The index represents PMBus rail when reading voltage / current, and
+    /// the sensor index when reading temperature (0-2) or fan speed (0-1).
+    index: u8,
+
+    mode: Cell<Option<pmbus::VOutModeCommandData>>,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Error {
+    BadRead { cmd: u8, code: ResponseCode },
+    BadWrite { cmd: u8, code: ResponseCode },
+    BadData { cmd: u8 },
+    BadValidation { cmd: u8, code: ResponseCode },
+    InvalidData { err: pmbus::Error },
+}
+
+impl From<Error> for ResponseCode {
+    fn from(err: Error) -> Self {
+        match err {
+            Error::BadRead { code, .. } => code,
+            Error::BadWrite { code, .. } => code,
+            Error::BadValidation { code, .. } => code,
+            _ => panic!(),
```

Comment:
This seems a little questionable in isolation. `From` seems like an easy impl to call without looking at the details; is a caller responsible for knowing the error is in one of the valid cases? Should this be `TryFrom` and let the caller panic?

Reply:
Good point – this matches previous code, but the previous code could probably be improved as well.  I switched to returning `ResponseCode::BadDeviceState` as a catch-all.